src/pages/index.js

```diff
@@ -158,50 +177,120 @@ closeButtons.forEach((button) => {
 });
 
+function handleImageClick(data) {
+  imagePreview.src = data.link;
+  imagePreviewCaption.textContent = data.name;
+  imagePreview.alt = data.name;
+  openModal(imageViewModal);
+}
+
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-button_liked");
+  api
+    .handleLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
 
-  closeModal(editModal);
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
 }
 
 editFormElement.addEventListener("submit", handleEditFormSubmit);
 
+editAvatarForm.addEventListener("submit", handleAvatarSubmit);
+
 newPostButton.addEventListener("click", () => {
   openModal(newPostModal);
 });
 
-// postCloseButton.addEventListener("click", () => {
-//   closeModal(newPostModal);
-// });
+postFormElement.addEventListener("submit", handleAddCard);
 
-// imageViewCloseButton.addEventListener("click", () => {
-//   closeModal(imageViewModal);
-// });
+deleteCardForm.addEventListener("submit", handleDeleteSubmit);
 
-postFormElement.addEventListener("submit", handleNewPostSubmit);
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .editAvatarInfo({ avatar: editAvatarInput.value })
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      closeModal(editAvatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
+}
 
-function handleNewPostSubmit(evt) {
+function handleAddCard(evt) {
   evt.preventDefault();
-  console.log(postLinkInput.value);
-  console.log(postCaptionInput.value);
-  const inputVariables = {
-    link: postLinkInput.value,
-    name: postCaptionInput.value,
-  };
-
-  const cardElement = getCardElement(inputVariables);
-  cardsList.prepend(cardElement);
-  closeModal(newPostModal);
-  evt.target.reset();
-  disableButton(cardSubmitButton, settings);
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .addCard({ name: postCaptionInput.value, link: postLinkInput.value })
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      closeModal(newPostModal);
+      postFormElement.reset();
```

Comment:
* You need to **`disable`** the submit button in the `todo form` if the inputs are **empty**. It’s needed because I can click the button and add an empty todo , but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty todo , the button should look **disabled** (gray) anyway and have a `disabled` attribute.


```js
      postFormElement.reset();
      disableButton(submitButton, settings)
```